### DISCUSSION TITLE
Why can't I render figures that are in a folder different from images?

### DISCUSSION THREAD

**Discussion by sbacelar** [quarto-dev/quarto-cli#8466] [Q&A] [answered]
### Description

I am using Windows 11 and Quarto version 1.4.549.
I have been trying to render some figures, but I had problems when the figures were not in the folder `images`, which is created (I think) when using Quarto in Visual mode. Should I copy all image files to the `images` folder to be able to render them?
In the code below, there is no problem with elephant figures, which are in the folder code/elephants; also there is no problem with my QQ1.png to QQ4.png files, which are in the folder code/images; but the other same files QQ1.png to QQ4.png that are in a longer path, do not render.
````qmd

---
title: "Teste Figs"
format: html
editor: source
execute:
  echo: false
---

```{r setup, include=FALSE}
library(tidyverse)
```

# Elephants

::: {layout-nrow="2"}
![Surus](elephants/surus.png)

![Hanno](elephants/hanno.png)

![Abdul Abbas](elephants/abdul-abbas.png)

![Lin Wang](elephants/lin-wang.png)
:::

# Q-Q Plots for each session in images

Some text.

::: {layout-nrow="2"}
![S1](images/QQ1.png)

![S2](images/QQ2.png)

![S3](images/QQ3.png)

![S4](images/QQ4.png)
:::

# Q-Q Plots for each session

Some text.

::: {layout-nrow="2"}
![S1](../data/out/figs/figs_T/QQ1.png)

![S2](../data/out/figs/figs_T/QQ2.png)

![S3](../data/out/figs/figs_T/QQ3.png)

![S4](../data/out/figs/figs_T/QQ4.png)
:::

````

**Comment by mcanouil** [comment]
Could you properly format your code and make fully reproducible?
We can't help you with your current message.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

![A placeholder image](https://placehold.co/600x400.png)

The end.
````
`````

**Comment by cscheid** [comment, marked as answer]
This is by design. Quarto is designed in part to encourage and facilitate reproducibility, and that includes enforcing care about the file-system dependencies in your document. If you want resource dependencies that move out of the current directory, you will need to create a Quarto project so that all required files are in the project directory.

**Reply by sbacelar** [reply]
I don't know if I explained myself well. My current directory is the project directory (.RProj). The structure of my project is similar to this:

```
.RProj
code/test.qmd
data/out/figs/some_fig.png
```
The problem is that for rendering to HTML, I must create a folder inside the code, like `code/figs/some_fig,png`. Is this right? 
Using the structure above, I see my figs in Visual mode, but when I try to render them, they do not appear in HTML.

**Reply by mcanouil** [reply]
Please share a small setup for example as a Git repository.
This will ease the understanding of your issue.

**Reply by sbacelar** [reply]
This is the address of the git repository I created: `https://github.com/sbacelar/quartoTest.git`.
Meanwhile, I think I solved (partially) the problem of rendering to HTML: at first, when I render the HTML page, it does not show the pictures in my browser (Chrome), but when I open the HTML file that has been created, the figures are there. Please try it with my GitHub code.

**Reply by mcanouil** [reply]
You need to add `_quarto.yml` to define the root of your project otherwise, it's the document, thus the images are outside the project.
Then you can use path relative to the root such as:

````qmd
::: {layout-nrow="2"}
![S1](/data/out/figs/QQ1.png)

![S2](/data/out/figs/QQ2.png)

![S3](/data/out/figs/QQ3.png)

![S4](/data/out/figs/QQ4.png)
:::
````

Note that Quarto recommends YAML code cell syntax:

````qmd
```{r}
#| label: setup
#| include: false
library(tidyverse)
```
````

instead of 

````qmd
```{r setup, include=FALSE}
library(tidyverse)
```
````

**Reply by sbacelar** [reply]
Thank you for your help. How can I use `_quarto.yml` to define the root of my project? Is it as simple as putting a yaml file called `_quarto.yml` at the root?

**Reply by mcanouil** [reply]
It can be empty in your case.

See <https://quarto.org/docs/projects/quarto-projects.html#shared-metadata>

**Reply by sbacelar** [reply]
I put an empty `_quarto.yml` at the root and the problem is solved. But when I render into `pdf` the pdf does not show immediately in the browser as it happened before. I have to open the pdf file.

**Reply by mcanouil** [reply]
How do you render/preview the Quarto documents? Are you using an IDE? Which one?
Please provide the steps to reproduce the behaviour your are describing.

**Reply by sbacelar** [reply]
I am using `RStudio` 2023.12.0 to render Quarto documents. I simply click on the Render button and choose the output type: HTML, PDF or Word. The YAML of my `qmd` file is now:
```
---
title: "Teste Figs"
editor: source
toc: true
highlight-style: pygments
format:
  html: 
    code-fold: true
    html-math-method: katex
  pdf:
    geometry: 
      - top=30mm
      - left=30mm
    include-before:
      - \newpage{}
    link-citations: true
  docx: default
execute:
  echo: false
---
```

**Reply by mcanouil** [reply]
This would be an issue with RStudio I believe.
I don't have RStudio so I can't reproduce, thus can't be sure.

**Reply by sbacelar** [reply]
OK. I will try later in CLI and VsCode.
